### DISCUSSION TITLE
Adding error table to keep track of exceptions and errors for C# port

### DISCUSSION THREAD

**Discussion by stishkin** [microsoft/onefuzz#2362] [Ideas]
Minor problem:

Currently we store all errors/exception in AppInsights

Check-pr uses that to validate that there were no issues in the run (it is also very ugly check, since there is bunch of filtering to avoid test failure due to Azure specific errors). Also adds "time delay" to validate whether there were issues

Writing tests that interact with the system - will also require accessing app insights to validate that there were no failures

I see one obvious solution :

Add azure table and write errors and exceptions in there
And then query the table (maybe even add REST API to the service to get that info from the service side)

**Comment by tevoinea** [comment]
I think this is a great idea, would it be only enabled for check-pr?

**Reply by stishkin** [reply]
Probably only enabled for check-pr

But if you have reasons to enable it in general - then I don't see reason why not to do it

**Reply by stishkin** [reply]
George mentioned that we might be able to manage with AppInsights data and there is no need for having table in the storage. (I'll investigate that further)
